src/SocialServer/Database/SocialDb.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Melia.Shared.Database;
+using MySql.Data.MySqlClient;
+using Yggdrasil.Security.Hashing;
+
+namespace Melia.Social.Database
+{
+	public class SocialDb : MeliaDb
+	{
+		/// <summary>
+		/// Returns account with given name, or null if it doesn't exist.
+		/// </summary>
+		/// <param name="name"></param>
+		/// <returns></returns>
+		public Account GetAccount(string name)
+		{
+			using (var conn = this.GetConnection())
+			using (var mc = new MySqlCommand("SELECT * FROM `accounts` WHERE `name` = @name", conn))
+			{
+				mc.Parameters.AddWithValue("@name", name);
+
+				using (var reader = mc.ExecuteReader())
+				{
+					if (!reader.Read())
+						return null;
+
+					var account = new Account();
+					account.Id = reader.GetInt64("accountId");
+					account.Name = reader.GetStringSafe("name");
+					account.TeamName = reader.GetStringSafe("teamName");
+					account.Password = reader.GetStringSafe("password");
+
+					// Upgrade MD5 hashes
```

Comment:
This shouldn't be necessary any longer, as we only used it to upgrade accounts that were created before we added BCrypt. Further, we already have it in Barracks, we should always be the first server the client connects to.